jbpm-human-task/jbpm-human-task-workitems/src/test/java/org/jbpm/services/task/wih/HTWorkItemHandlerBaseTest.java

```diff
@@ -129,6 +129,80 @@ public void testTaskMultipleActors() throws Exception {
+
+        TestWorkItemManager manager = new TestWorkItemManager();
+        ksession.setWorkItemManager(manager);
+        WorkItemImpl workItem = new WorkItemImpl();
+        workItem.setName("Human Task");
+        workItem.setParameter("NodeName", "TaskName");
+        workItem.setParameter("Comment", "Comment");
+        workItem.setParameter("Priority", "10");
+        workItem.setParameter("ActorId", "Darth Vader, Dalai Lama");
+        workItem.setProcessInstanceId(10);
+        getHandler().executeWorkItem(workItem, manager);
+
+        List<TaskSummary> tasks = taskService.getTasksAssignedAsPotentialOwner("Darth Vader", "en-UK");
+        assertEquals(1, tasks.size());
+        TaskSummary task = tasks.get(0);
+        assertEquals("TaskName", task.getName());
+        assertEquals(10, task.getPriority().intValue());
+        assertEquals("Comment", task.getSubject());
+        assertEquals("Comment", task.getDescription());
+        assertEquals(Status.Ready, task.getStatus());
+
+        taskService.claim(task.getId(), "Darth Vader");
+
+        taskService.start(task.getId(), "Darth Vader");
+
+        taskService.complete(task.getId(), "Darth Vader", null);
+
+        assertTrue(manager.waitTillCompleted(MANAGER_COMPLETION_WAIT_TIME));
+
+        System.clearProperty("org.jbpm.ht.copy.task_subject.to.task_description");
+    }
+
+    @Test
+    public void testTaskDescriptinWithoutCopyingFromTaskSubject() throws Exception {
+
+        System.setProperty("org.jbpm.ht.copy.task_subject.to.task_description", "false");
+
+        TestWorkItemManager manager = new TestWorkItemManager();
+        ksession.setWorkItemManager(manager);
+        WorkItemImpl workItem = new WorkItemImpl();
+        workItem.setName("Human Task");
+        workItem.setParameter("NodeName", "TaskName");
+        workItem.setParameter("Comment", "Comment");
+        workItem.setParameter("Description", "Description");
+        workItem.setParameter("Priority", "10");
+        workItem.setParameter("ActorId", "Darth Vader, Dalai Lama");
+        workItem.setProcessInstanceId(10);
+        getHandler().executeWorkItem(workItem, manager);
+
+        List<TaskSummary> tasks = taskService.getTasksAssignedAsPotentialOwner("Darth Vader", "en-UK");
+        assertEquals(1, tasks.size());
+        TaskSummary task = tasks.get(0);
+        assertEquals("TaskName", task.getName());
+        assertEquals(10, task.getPriority().intValue());
+        assertEquals("Comment", task.getSubject());
+        assertEquals("Description", task.getDescription());
+        assertEquals(Status.Ready, task.getStatus());
+
+        taskService.claim(task.getId(), "Darth Vader");
+
+        taskService.start(task.getId(), "Darth Vader");
+
+        taskService.complete(task.getId(), "Darth Vader", null);
+
+        assertTrue(manager.waitTillCompleted(MANAGER_COMPLETION_WAIT_TIME));
+
+        System.clearProperty("org.jbpm.ht.copy.task_subject.to.task_description");
```

Comment:
You can refactor and extract this logic to a private method, instead of duplicating this logic, and just setting the system property with true or false, for each one, so you avoid duplicating the same logic

Reply:
As per our discussion over chat keeping these method as is